### ISSUE TITLE
add captcha/email confirmation to the signup workflow

### ISSUE THREAD

**Issue by silky** [scirate/scirate#378]

**Comment by JacksonBates** [comment]
Is this still required? There is a confirmation / activation email upon signup. Mine went to spam - but it was easy enough to find.

Is there a problem with spam accounts?

**Comment by silky** [comment]
hey @JacksonBates 

yes; this would be a hugely useful item!

there's around 40,000 spam accounts at the moment; with a few hundred added each week.

**Comment by JacksonBates** [comment]
Do you have any preferences?

Google reCaptcha is pretty standard, and their v3 claims to still be effective while getting out of the user's way - probably at the cost of a bunch of JS and privacy though!

Happy to investigate others or implement one of the Google ones.

**Comment by silky** [comment]
no, no preference at all; whatever you think is best/easiest sounds great!

**Comment by JacksonBates** [comment]
Awesome - I'll probably start on this one next.

**Comment by JacksonBates** [comment]
I just had to sign up for a trial of something at work and I noticed they were using the v3 captcha behind the scenes...seamless experience! I'm going to implement that instead of v2 :)

**Comment by silky** [comment]
sounds great :)

**Comment by JacksonBates** [comment]
I've got this working, but it needs some clean up.

Quick question - do you keep all the env variables in `local_secrets.rb` for this project, so if I reference `Settings::RECAPTCHA_SECRET_KEY` in the controller that'll be consistent with the rest of the project?

**Comment by silky** [comment]
yes, @JacksonBates , you're exactly right! didn't even know that's how we did it :)

thanks for this! really looking forward to having this done; it's going to be so useful

**Comment by JacksonBates** [comment]
I moved the keys into local settings and then it didn't work...have to have another crack at this this evening :)

**Comment by silky** [comment]
on the server there is just `local_settings.rb` in the root folder and it looks like this:

```
module LocalSettings
  HOST = 'scirate.com'
  # Other secrets ...
end
```

**Comment by silky** [comment]
fixed by @JacksonBates in #391 #391
